packages/@uppy/companion/src/server/Uploader.js

```diff
@@ -551,6 +568,55 @@ class Uploader {
       this.cleanUp()
     })
   }
+
+  /**
+   * Upload a stream to B2 while it is still being downloaded.
+   */
+  uploadB2 () {
+    if (!this.options.b2) {
+      this.emitError(new Error('The B2 client is not configured on this companion instance.'))
+      return
+    }
+    const filename = this.options.metadata.filename || path.basename(this.path)
+    const fileSize = this.options.size
+    const { client, options } = this.options.b2
+
+    if (!this._b2EndpointPool) {
+      this._b2EndpointPool = client.createEndpointPool(options.bucket)
+    }
+
+    const upload = client.upload({
+      bucketName: options.bucket,
+      path: this.path,
+      fileName: options.getPath(null, filename), // destination
+      fileSize, // expected final size of the file being transferred
+      stream: this.beginTailReadStream(),
+      endpointPool: this._b2EndpointPool
+    })
+
+    this.b2Upload = upload
+
+    upload.onUploadProgress = ({ loaded, total }) => {
+      this.emitProgress(loaded, total)
+    }
+
+    upload.send((error, data) => {
+      this.b2Upload = null
+      if (error) {
+        this.emitError(error)
+      } else {
+        this.emitSuccess('?', {
```

Comment:
Ah yes, I should have asked about this. If the destination bucket is public, `b2_download_file_by_id` can be used to get a download URL, but that call is a class B transaction so it comes with a small fee.

Perhaps I expose this as a configuration option? When enabled, returns a valid download URL, making it functionally the same as the s3 response, or `null` if `getDownloadURL` (or some better name) is disabled?

Reply:
Another option may be returning a thunk to make the API call to obtain the download URL

Reply:
> Perhaps I expose this as a configuration option? When enabled, returns a valid download URL, making it functionally the same as the s3 response, or null if getDownloadURL (or some better name) is disabled?

I think we can with this option, yes. It gives the server control over whether or not to allow its usage (since it's costly) 👍